doc/man1/genpkey.pod

```diff
@@ -160,91 +156,92 @@ If set the key is restricted and B<len> specifies the minimum salt length.
 
 =back
 
-=head1 DSA PARAMETER GENERATION OPTIONS
+=head2 EC Key Generation Options
+
+The EC key generation options can also be used for parameter generation.
 
 =over 4
 
-=item B<dsa_paramgen_bits:numbits>
+=item B<ec_paramgen_curve:curve>
 
-The number of bits in the generated parameters. If not specified 1024 is used.
+The EC curve to use. OpenSSL supports NIST curve names such as "P-256".
+
+=item B<ec_param_enc:encoding>
+
+The encoding to use for parameters. The "encoding" parameter must be either
+"named_curve" or "explicit". The default value is "named_curve".
 
 =back
 
-=head1 DH PARAMETER GENERATION OPTIONS
+=head1 PARAMETER GENERATION OPTIONS
+
+The options supported by each algorithm and indeed each implementation of an
+algorithm can vary. The options for the OpenSSL implementations are detailed
+below.
+
+=head2 DSA Parameter Generation Options
 
 =over 4
 
-=item B<dh_paramgen_prime_len:numbits>
+=item B<dsa_paramgen_bits:numbits>
 
-The number of bits in the prime parameter B<p>.
+The number of bits in the generated prime. If not specified 1024 is used.
 
-=item B<dh_paramgen_generator:value>
+=item B<dsa_paramgen_q_bits:numbits>
 
-The value to use for the generator B<g>.
+The number of bits in the q parameter. Must be one of 160, 224 or 256. If not
+specified 160 is used.
 
-=item B<dh_rfc5114:num>
+=item B<dsa_paramgen_md:digest>
 
-If this option is set then the appropriate RFC5114 parameters are used
-instead of generating new parameters. The value B<num> can take the
-values 1, 2 or 3 corresponding to RFC5114 DH parameters consisting of
-1024 bit group with 160 bit subgroup, 2048 bit group with 224 bit subgroup
-and 2048 bit group with 256 bit subgroup as mentioned in RFC5114 sections
-2.1, 2.2 and 2.3 respectively.
+The digest to use during parameter generation. Must be one of sha1, sha224 or
```

Comment:
Are params case insensitive?  If not, can we change that?

Reply:
Also, since it's literal words, sha1 etc should be in quotes.  Like the curve name above.  This might have other effects outside the context diff.

Reply:
Some experimentation reveals that we accept sha224 or SHA224 but not other variants (e.g. Sha224). Same for the other digests. Do we need to document that?

Reply:
I think we just document what we officially accept, and make sure to use markup to indicate that they are literal words.